src/test/java/ReaderWriterTest.java

```diff
@@ -0,0 +1,91 @@
+package test;
+
+import IO.DbReader;
+import IO.DbWriter;
+import data.Database;
+import data.DatabaseQuery;
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.OutputStream;
+import java.io.Reader;
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+/* Validate that a .txt file can be properly be stored written and read from database.  */
+public final class ReaderWriterTest {
```

Comment:
Looks good to me, only suggestion is to add more comments specifically in this file and try to break down the whole process of writing and reading. Minor things like explaining why you included the setup method with nothing in it, or why you created a second reader and writer while hardly using the first. This would help me process the whole file a lot better.